### PR TITLE
2350 naruto run animation has scale transforms

### LINKED ISSUE #2350: Naruto run animation has scale transforms
The naruto run animation is violating the standard by having scale transforms on the animation.

```
$ node animations-baker.js
```

```
unknown track name naruto run.fbx VectorKeyframeTrack {
  name: 'mixamorigRightHandRing1.scale',
  times: Float32Array(2) [ 0, 0.4000000059604645 ],
  values: Float32Array(6) [
    0.9999999403953552,
    0.9999999403953552,
    1,
    0.9999999403953552,
    0.9999999403953552,
    1
  ],
  createInterpolant: [Function: InterpolantFactoryMethodLinear]
}
```

### PR DESCRIPTION
**Pull request by 0reo**
This PR ensures that any animations with animation tracks applying scaling throw an error.  Animation tracks with scales all set to 1 are just removed.  Closes #2350.  Disable white space when reviewing.

as per https://github.com/webaverse/app/issues/2350#issuecomment-1074449065

> Failing that I'm ok with just adding to our code a loop that checks that if there is a scale, it must be all = 1, and then it can be thrown away (if that requirement is violated we should crash because there is currently no way we can render such an animation).

### CODE REVIEW

File: animations-baker.js [hunk 1]
```diff
@@ -344,10 +345,23 @@ const baker = async (uriPath = '', fbxFileNames, vpdFileNames, outFile) => {
           // const bone = bones.find(b => b.name === boneName);
           const boneInterpolant = new THREE.QuaternionLinearInterpolant(track.times, track.values, track.getValueSize());
           boneQuaternionInterpolants[boneName] = boneInterpolant;
+        } else if (/\.scale$/.test(track.name)) {
+          if (allOnes(track.values)) {
+            const index = tracks.indexOf(track);
+            if (index !== -1) {
```
Comment (avaer) [inline animations-baker.js:351]: Why would this ever not be the case?

File: animations-baker.js [hunk 2]
```diff
@@ -344,10 +345,23 @@ const baker = async (uriPath = '', fbxFileNames, vpdFileNames, outFile) => {
...
+            }
+          } else {
+            throw new Error(`Error with the following track.  All scale transforms must be set to 1. Aborting.\n Animation: ${animation.name}, Track: ${track.name}, values: \n ${track.values}`);
+          }
         } else {
           console.warn('unknown track name', animation.name, track);
         }
       }
+      // remove scale transform tracks as they won't be used;
+      tracksToRemove.forEach(i => {
```
Comment (avaer) [inline animations-baker.js:362]: I don't think this is legal/correct for all combinations. Splicing changes the index that you must splice for other items.

File: animations-baker.js [hunk 3]
```diff
@@ -330,9 +329,11 @@ const baker = async (uriPath = '', fbxFileNames, vpdFileNames, outFile) => {
       const rootBone = object; // not really a bone
       const leftFootBone = bones.find(b => b.name === 'mixamorigLeftFoot');
       const rightFootBone = bones.find(b => b.name === 'mixamorigRightFoot');
+      const allOnes = arr => arr[0] === 1 && arr.every(v => v === arr[0]);
```
Comment (avaer) [inline animations-baker.js:332]: What a weird check. Why is the first item special? If you want to know there is at least one item, checking `.length` is better.

**Review by avaer — CHANGES_REQUESTED** [review]
I think the code is wrong.

File: animations-baker.js (hunk 2 above)
Comment (0reo) [reply animations-baker.js:362]: yes you're right, will adjust

File: animations-baker.js (hunk 1 above)
Comment (0reo) [reply animations-baker.js]: removed redundant check

File: animations-baker.js (hunk 3 above)
Comment (0reo) [reply animations-baker.js]: removed redundant check

File: animations-baker.js (hunk 2 above)
Comment (0reo) [reply animations-baker.js]: corrected to go backwards, to prevent changing index's for values which need to be removed

**Pull request merged**